Explorer/Assets/DCL/SDKComponents/Tween/Systems/WriteTweenStateSystem.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using CRDT;
+using CrdtEcsBridge.ECSToCRDTWriter;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using ECS.Abstract;
+using ECS.LifeCycle;
+using ECS.LifeCycle.Components;
+using ECS.Unity.Groups;
+using ECS.Unity.Tween.Components;
+using System;
+
+namespace ECS.Unity.Tween.Systems
+{
+    /// <summary>
+    ///     Utility to write Tween State to CRDT (propagate it back to the scene)
+    /// </summary>
+    [UpdateInGroup(typeof(ComponentInstantiationGroup))]
+    [UpdateAfter(typeof(TweenUpdaterSystem))]
+    [LogCategory(ReportCategory.TWEEN)]
+    public partial class WriteTweenStateSystem : BaseUnityLoopSystem, IFinalizeWorldSystem
+    {
+        private readonly IECSToCRDTWriter ecsToCRDTWriter;
+
+        public WriteTweenStateSystem(World world, IECSToCRDTWriter ecsToCRDTWriter)
+            : base(world)
+        {
+            this.ecsToCRDTWriter = ecsToCRDTWriter;
+        }
+
+        protected override void Update(float t)
+        {
+            ExecuteQuery(World);
+            RemoveQuery(World);
+        }
+
+        [Query]
+        private void Execute(ref CRDTEntity sdkEntity, ref SDKTweenComponent tweenComponent)
+        {
+            if (!tweenComponent.IsTweenStateDirty) return;
```

Comment:
I believe this flag you don't need either. You can just call this logic from `TweenUpdaterSystem`. You can encapsulate methods connected to the writing in a separate static utility for example.
I'd aim at simplification here.

Reply:
I agree that you could have the `ecsToCRDTWriter.PutMessage` of `PBTweenState` in `TweenUpdaterSystem`, and also the `Remove<PBTweenState>()`.
Thus removing this `WriteTweenStateSystem` entirely.

Reply:
good point! nice 👍